### PR TITLE
Modernise the plugin

### PR DESCRIPTION
**Pull request by jmMeessen**
# Modernising the plugin

(Following the Hacktoberfest 2021 recommendations)

* Update parent pom to 4.29
* Update the BOM to bom-2.289.x
* Update test dependencies versions
* Enable incremental
* Enable Continuous Delivery

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did

### CODE REVIEW

File: .mvn/maven.config [hunk 1]
```diff
@@ -0,0 +1,3 @@
+-Pconsume-incrementals
+-Pmight-produce-incrementals
+-Dchangelist.format=%d.v%s
```
Comment (jglick) [inline .mvn/maven.config:3]: This is to set up CD. Have you done so?

File: .mvn/maven.config [hunk 2]
```diff
@@ -0,0 +1,3 @@
+-Pconsume-incrementals
+-Pmight-produce-incrementals
```
Comment (jglick) [inline .mvn/maven.config:2]: This is to set up Incrementals. You have not done that: https://github.com/jenkinsci/webhook-step-plugin/blob/a707a7cbebe08f2dae1e08d7bf8ae716472f1511/pom.xml#L13

File: .mvn/gitpod-mvn-settings.xml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
+          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+          xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
+                              http://maven.apache.org/xsd/settings-1.0.0.xsd">
+  <localRepository>/workspace/m2-repository/</localRepository>
+  <pluginGroups>
+    <pluginGroup>org.jenkins-ci.tools</pluginGroup>
+  </pluginGroups>
+
+  <profiles>
```
Comment (jglick) [inline .mvn/gitpod-mvn-settings.xml:10]: Meant for #22?

File: .gitpod.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+tasks:
```
Comment (jglick) [inline .gitpod.yml:1]: Ditto?

File: pom.xml [hunk 5]
```diff
@@ -5,22 +5,25 @@
...
 
   <properties>
-    <jenkins.version>2.138.4</jenkins.version>
+    <jenkins.version>2.289.1</jenkins.version>
     <java.level>8</java.level>
+    <gitHubRepo>jenkinsci/webhook-step-plugin</gitHubRepo>
   </properties>
 
+  <!-- remove me -->
+
```
Comment (jglick) [inline pom.xml:23]: ?
```suggestion
```

File: pom.xml [hunk 6]
```diff
@@ -5,22 +5,25 @@
...
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>3.49</version>
+    <version>4.29</version>
     <relativePath />
   </parent>
-  <groupId>org.jenkins-ci.plugins</groupId>
+
   <artifactId>webhook-step</artifactId>
   <version>1.5-SNAPSHOT</version>
```
Comment (jglick) [inline pom.xml:13]: So https://github.com/jenkinsci/webhook-step-plugin/blob/30e3cb0c455e05bcadf5fe5f03ce0c74fdd3929b/.mvn/maven.config#L1-L3 and https://github.com/jenkinsci/webhook-step-plugin/blob/30e3cb0c455e05bcadf5fe5f03ce0c74fdd3929b/.mvn/extensions.xml#L4 are still unused.

Comment (jglick) [reply pom.xml:13]: see https://www.jenkins.io/doc/developer/plugin-development/incrementals/ + https://www.jenkins.io/doc/developer/publishing/releasing-cd/

Comment (jmMeessen) [reply pom.xml]: Thank you to point me to this, @jglick. I learned something new. (although, still scratching my head and trying to be sure I understood it correctly)
Will study this (and the CD process) more in details.

TIL: don't copy configurations from another project without understanding enough of what they mean...

Comment (jglick) [reply pom.xml]: And remember that without `cd.enabled = true` in RPU you will not be able to deploy.

File: pom.xml [hunk 7]
```diff
@@ -5,16 +5,19 @@
...
 
   <artifactId>webhook-step</artifactId>
-  <version>1.5-SNAPSHOT</version>
+  <version>${revision}${changelist}</version>
   <packaging>hpi</packaging>
 
   <properties>
-    <jenkins.version>2.138.4</jenkins.version>
+    <revision>1.5</revision>
+    <changelist>-SNAPSHOT</changelist>
```
Comment (jglick) [inline pom.xml:18]: Still contradicts https://github.com/jenkinsci/webhook-step-plugin/blob/0f9d072607b68eb16cd16920c4469fc9598d27d2/.mvn/maven.config#L3

File: .github/workflows/CD.yaml [hunk 8]
```diff
@@ -0,0 +1,60 @@
+# Note: additional setup is required, see https://www.jenkins.io/redirect/continuous-delivery-of-plugins
+
+name: cd
+on:
+  workflow_dispatch:
+  # Commenting out to deploy explicitly (not on every push on master)
```
Comment (jglick) [inline .github/workflows/CD.yaml:6]: Note that the default is deploy only on merges of PRs with specific labels.

**Pull request merged**